tests/std/tests/P3349R1_contiguous_iterators_to_pointers/test.cpp

```diff
@@ -0,0 +1,463 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <format>
+#include <functional>
+#include <iterator>
+#include <memory>
+#include <numeric>
+#include <ranges>
+#include <string>
+#include <vector>
+
+using namespace std;
+
+struct safe_iter_out_of_bounds_err {};
+
+enum range_type { range_large = 3, range_medium = 4, range_small = 5 };

```

Comment:
`range_large` and `range_small` are currently unused, but I guess it's okay to keep them. However, I recommend upgrading this to an `enum class`, which is more type-safe. (The less we have to worry about weird stuff happening in test code, the more time we have to worry about the weird stuff in product code.)